pkg/util/fedinformer/transform.go

```diff
@@ -0,0 +1,60 @@
+package fedinformer
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// StripUnusedFields is the transform function for shared informers,
+// it removes unused fields from objects before they are stored in the cache to save memory.
+func StripUnusedFields(obj interface{}) (interface{}, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		// shouldn't happen
+		return obj, nil
+	}
+	// ManagedFields is large and we never use it
+	accessor.SetManagedFields(nil)
+	return obj, nil
+}
+
+// StripUnusedFieldsForNode is the transform function for shared informers,
+// it removes unused fields from nodes before they are stored in the cache to save memory.
+func StripUnusedFieldsForNode(obj interface{}) (interface{}, error) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		// shouldn't happen
+		return obj, nil
+	}
+
+	aggregatedNode := &corev1.Node{
+		Status: corev1.NodeStatus{
+			Allocatable: node.Status.Allocatable,
+			Conditions:  node.Status.Conditions,
+		},
+	}
+	return aggregatedNode, nil
+}
+
+// StripUnusedFieldsForPod is the transform function for shared informers,
+// it removes unused fields from pods before they are stored in the cache to save memory.
+func StripUnusedFieldsForPod(obj interface{}) (interface{}, error) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		// shouldn't happen
+		return obj, nil
+	}
+
+	aggregatedPod := &corev1.Pod{
+		Spec: corev1.PodSpec{
```

Comment:
👍Nice, I guess this is why we are using another **typed informer** so that it won't impact other places where we fetch pods from cache

Reply:
Yes, but not exactly. `typed informer` has better performance than `generic informer`. Yes, we still benefit from that because here we can safely use the transform function.